x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/SemanticSearchAction.java

```diff
@@ -0,0 +1,435 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.ml.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.index.query.AbstractQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.search.fetch.StoredFieldsContext;
+import org.elasticsearch.search.fetch.subphase.FetchSourceContext;
+import org.elasticsearch.search.fetch.subphase.FieldAndFormat;
+import org.elasticsearch.search.vectors.KnnSearchBuilder;
+import org.elasticsearch.xcontent.ObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xpack.core.ml.inference.trainedmodel.TextEmbeddingConfigUpdate;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+public class SemanticSearchAction extends ActionType<SemanticSearchAction.Response> {
+
+    public static final String NAME = "cluster:monitor/xpack/ml/semantic_search"; // TODO what should this be called?
```

Comment:
I agree it's not desirable. This is effectively just calling `_infer` and then `_search`. Meaning, a user calling this API could negatively effect other users calling `_infer` by filling up the inference queue and using those resources. So, giving it permissions that are not the same as the `_infer` API could be considered a security loop-hole. 

Maybe this indicates a problem with the current security level of `_infer` as a whole.

Reply:
We discussed this in the NLP sync meeting yesterday.

The consensus was that there _wouldn't_ be a need to allow some users to do normal searches against an index but _not_ semantic searches.  Therefore, use of this endpoint should be granted by the `read` index privilege on the indices being searched.

The action name should probably then be named `indices:data/read/semantic_search`.  Even if it's not that it must match the pattern `indices:data/read/*`.  Then when we execute `InferTrainedModelDeploymentAction` we should use the internal `_xpack` user with `ML_ORIGIN` so that it doesn't require extra privileges over and above what the outer action requires.

If for some reason that's really hard to do in this PR, for example if it requires a change in the security plugin, then I don't mind merging this PR with the current privilege and changing to an index privilege in a followup that will be easier for the security team to review.  But this must be the very next thing that is worked on so that there's no danger of us shipping 8.6 with the wrong action name here.  The `async-search` plugin is also an X-Pack plugin that has actions that match `indices:data/read/*`, so that can be used as an example of how to have index actions in a plugin.

Issues with accidental or deliberate denial of service against the inference processes do need to be considered eventually, but not as part of this PR or the way the endpoint is authorized. An idea was to have some sort of throttling per user or per client or something. But this can be discussed in a separate issue.

Reply:
Changing the action to a `indices:data/read/...` action does require a change in the security plugin although that change may be quite simple. However, I would not like to conflate those changes, the security change should be made in a dedicated PR and we can proceed with this feature PR at the higher required privilege and change it shortly after in a follow up PR.